cpp/memilio/epidemiology/damping.h

```diff
@@ -519,16 +519,15 @@ class Dampings
     static void inclusive_exclusive_sum_rec(Iter b, Iter e, Matrix& sum)
     {
         if (b != e) {
-            sum = sum + std::get<Matrix>(*b) - (sum.array() * std::get<Matrix>(*b).array()).matrix();
+            sum = (sum + std::get<Matrix>(*b) - (sum.array() * std::get<Matrix>(*b).array()).matrix()).eval();
             inclusive_exclusive_sum_rec(++b, e, sum);
         }
     }
     template <class Tuple>
     static Matrix inclusive_exclusive_sum(const std::vector<Tuple>& v)
     {
         assert(!v.empty());
-        auto& m  = std::get<Matrix>(v.front());
-        auto sum = m.eval();
+        Matrix sum = std::get<Matrix>(v.front());
```

Comment:
note to self, sometimes obsessive use of `auto` leads to less readable code. your version is clearer but equivalent.